src/engine/sidechain/enginebroadcast.cpp

```diff
@@ -105,712 +93,106 @@ EngineBroadcast::~EngineBroadcast() {
-        if (queuelen > kMaxNetworkCache) {
-            m_lastErrorStr = tr("Network cache overflow");
-            tryReconnect();
-        }
-    }
-}
-// These are not used for streaming, but the interface requires them
-int EngineBroadcast::tell() {
-    if (!m_pShout) {
-        return -1;
-    }
-    return -1;
-}
-// These are not used for streaming, but the interface requires them
-void EngineBroadcast::seek(int pos) {
-    Q_UNUSED(pos)
-    return;
-}
-// These are not used for streaming, but the interface requires them
-int EngineBroadcast::filelen() {
-    return 0;
-}
-
-bool EngineBroadcast::writeSingle(const unsigned char* data, size_t len) {
-    // We are already synced by EngineNetworkstream
-    setFunctionCode(8);
-    int ret = shout_send_raw(m_pShout, data, len);
-    if (ret == SHOUTERR_BUSY) {
-        // in case of busy, frames are queued
-        // try to flush queue after a short sleep
-        qDebug() << "EngineBroadcast::writeSingle() SHOUTERR_BUSY, trying again";
-        QThread::msleep(10); // wait 10 ms until "busy" is over. TODO() tweak for an optimum.
-        // if this fails, the queue is transmitted after the next regular shout_send_raw()
-        (void)shout_send_raw(m_pShout, nullptr, 0);
-    } else if (ret < SHOUTERR_SUCCESS) {
-        m_lastErrorStr = shout_get_error(m_pShout);
-        qDebug() << "EngineBroadcast::writeSingle() error:"
-                 << ret << m_lastErrorStr;
-        NetworkStreamWorker::debugState();
-        if (++m_iShoutFailures > kMaxShoutFailures) {
-            tryReconnect();
-        }
-        return false;
-    } else {
-        m_iShoutFailures = 0;
-    }
-    return true;
-}
-
 void EngineBroadcast::process(const CSAMPLE* pBuffer, const int iBufferSize) {
     setFunctionCode(4);
 
-    setState(NETWORKSTREAMWORKER_STATE_BUSY);
-    // If we are here then the user wants to be connected (broadcast is enabled
-    // in the preferences).
-
-    // If we aren't connected, bail.
-    if (m_iShoutStatus != SHOUTERR_CONNECTED)
-        return;
-
-    // If we are connected, encode the samples.
-    if (iBufferSize > 0 && m_encoder) {
-        setFunctionCode(6);
-        m_encoder->encodeBuffer(pBuffer, iBufferSize);
-        // the encoded frames are received by the write() callback.
-    }
-
-    // Check if track metadata has changed and if so, update.
-    if (metaDataHasChanged()) {
-        updateMetaData();
-    }
-    setState(NETWORKSTREAMWORKER_STATE_READY);
-}
-
-bool EngineBroadcast::metaDataHasChanged() {
-    TrackPointer pTrack;
-
-    // TODO(rryan): This is latency and buffer size dependent. Should be based
-    // on time.
-    if (m_iMetaDataLife < 16) {
-        m_iMetaDataLife++;
-        return false;
-    }
-
-    m_iMetaDataLife = 0;
-
-    pTrack = PlayerInfo::instance().getCurrentPlayingTrack();
-    if (!pTrack)
-        return false;
+    QList<ShoutOutputPtr> connections = m_connections.values();
+    for(ShoutOutputPtr c : connections) {
```

Comment:
> I think "connection" is better to differentiate it from sound card outputs.

I agree, this is a more fitting word. When creating this class I was unsure about the terminology used in multi-broadcasting, but I've figured out so far that we can internally refer to two things:
- A profile: the "definition" of a Live Broadcasting connection
- A connection: actual libshout connection to a streaming server based on settings provided by a profile.

And I believe users only have to know about the term "connection" because the file nature (with possible import/export) of profiles isn't exposed to them, so "profiles" in the Live Broadcasting settings are abstracted to "connections".